### PR TITLE
Allow 11 digits only when a phone number starts with 1800

### LINKED ISSUE #4365: Allow 11 digits only when a phone number starts with 18001
*Bug*
Currently, phone inputs are allowing 11-digit numbers even though it does not start with 18001

*Expected*
Only allow 11 digits when it starts with 18001 otherwise make it 10 digits.

follow up for https://github.com/coronasafe/care_fe/pull/4294

cc: @nihal467 @skks1212

### PR DESCRIPTION
**Pull request by khavinshankar**
## Proposed Changes

- Fixes #4365 
- Allow 11 digits only when a phone number starts with 18001

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [x] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [x] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Review by skks1212 — APPROVED** [review]

**Comment by nihal467** [comment]
@khavinshankar 
image

https://deploy-preview-4366--care-net.netlify.app/facility/7ab27974-32c8-4e36-bdf9-dbf102524718/assets/ab12b691-5769-4e6f-a6d1-d6d2028dca07/update

unable to input 11 digits into the asset form

File: src/Components/Common/HelperInputFields.tsx [hunk 1]
```diff
@@ -653,16 +653,21 @@ export const PhoneNumberField = (props: any) => {
...
   const onChangeHandler = debounce(onChange, 500);
+
   const handleChange = (
     value: string,
     data: Partial<ICountryData>,
     event: ChangeEvent<HTMLInputElement>,
     formattedValue: string
   ) => {
     onChangeHandler(formattedValue);
+    setMaxLength(() => (value.slice(2, 7) === "18001" ? 16 : 15));
```
Comment (khavinshankar) [inline src/Components/Common/HelperInputFields.tsx]:
```suggestion
    setMaxLength(() => (value.slice(2, 6) === "1800" ? 16 : 15));
```

**Comment by nihal467** [comment]
@khavinshankar 
image

error persist, unable to input 11 digit in the platform

**Comment by nihal467** [comment]
LGTM

**Pull request merged**